x-pack/plugins/task_manager/server/routes/health.ts

```diff
@@ -114,7 +115,8 @@ export function healthRoute(params: HealthRouteParams): {
       }),
       // Only calculate the summarized stats (calculates all running averages and evaluates state)
       // when needed by throttling down to the requiredHotStatsFreshness
-      map((stats) => withServiceStatus(getHealthStatus(stats)))
+      map((stats) => withServiceStatus(getHealthStatus(stats))),
+      filter(([monitoredHealth]) => !monitoredHealth.isEmpty)
```

Comment:
@ymao1 I haven't checked the impact of filtering at this level, not sure we want the `monitoredHealth$` observable to have the "empty" events.

Reply:
Looking at the code, all the subscribers to the observable are trying to pull various stats from the `monitoredHealth$.stats` which would be empty anyway in this case, so I believe the impact is small. I'm unable to run it locally right now (poor internet connection) but will give it a go first thing next week.